apps/meteor/server/features/EmailInbox/EmailInbox_Incoming.ts

```diff
@@ -127,14 +127,19 @@ async function uploadAttachment(attachment: Attachment, rid: string, visitorToke
 }
 
 export async function onEmailReceived(email: ParsedMail, inbox: string, department = ''): Promise<void> {
-	logger.debug(`New email conversation received on inbox ${inbox}. Will be assigned to department ${department}`);
+	logger.debug(`New email conversation received on inbox ${inbox}. Will be assigned to department ${department}`, email);
 	if (!email.from?.value?.[0]?.address) {
 		return;
 	}
 
 	const references = typeof email.references === 'string' ? [email.references] : email.references;
+	const initialRef = [email.messageId, email.inReplyTo].filter((t): boolean => !!t) as string[];
+	// thread = references.flatmap(t => t.split(','))
```

Comment:
:eyes:

Reply:
yeah, [string|undefined] => [string], is it too ugly?

Reply:
no no, i was referring to the comment haha

On the other one, you can use `filter(Boolean)` and that should also work

Reply:
it's what i wanted to write, it's supported on browsers, maybe one day we'll get it on node, then we can delete the ugly code.